src/matrix_content_scanner/scanner/scanner.py

```diff
@@ -478,13 +481,10 @@ def _run_scan(self, file_name: str) -> int:
         Returns:
             The exit code the script returned.
         """
-        try:
-            subprocess.run([self._script, file_name], check=True)
-            logger.info("Scan succeeded")
-            return 0
-        except subprocess.CalledProcessError as e:
-            logger.info("Scan failed with exit code %d: %s", e.returncode, e.stderr)
-            return e.returncode
+        async with self._current_scan_semaphore:
+            process = await asyncio.create_subprocess_exec(self._script, file_name)
+            retcode = await process.wait()
```

Comment:
`process.wait()` catches errors now and gives an error code instead of manually doing it?

Reply:
That's right. In a little more detail: [`CalledProcessError`](https://docs.python.org/3/library/subprocess.html?highlight=subprocess%20run#subprocess.CalledProcessError) is only raised when `check=True` and the subprocess returns something nonzero. So the return code handling should be the same. For completeness, [`wait` is described here](https://docs.python.org/3/library/asyncio-subprocess.html#asyncio.subprocess.Process.wait).

Under the hood, I _think_ they're doing the same thing (invoking subprocess.Popen); it's just that the two functions expose the error case in two different ways. 

Having just looked at this, I notice that I've unintentionally removed the logging; let me add that back in.

Reply:
Maybe worth adding the filename or something to that line too?

Reply:
I tried this, but you get the full file path on disk which is a bit verbose. What you probably want is the media ID; you can dig this up from logs by following the request ID.

I'm inclined to leave this as it was, unless you feel strongly?

Reply:
I've added this back in. I sanity checked my stderr reading and logging worked by
- running the perf script with `-v`, to show scanner logs; and by
- making 5% of dummy scans randomly fail.

```
$ python perf/scanner_perf.py -v 2>&1 | grep 'Scanning failed' | head -n 3
2023-11-28 18:43:31,116 - matrix_content_scanner.scanner.scanner - 493 - INFO - GET-29 - Scanning failed with exit code 1. Stderr: I don't like the look of /home/dmr/workspace/matrix-content-scanner-python/perf/temp/element.io/6130836e26b462a6fe63d4e080dd9d2037490f2b
2023-11-28 18:43:32,578 - matrix_content_scanner.scanner.scanner - 493 - INFO - GET-42 - Scanning failed with exit code 1. Stderr: I don't like the look of /home/dmr/workspace/matrix-content-scanner-python/perf/temp/hackerlab.in/vjENMlrncPUGDmbyMZhWJzkG
2023-11-28 18:43:32,907 - matrix_content_scanner.scanner.scanner - 493 - INFO - GET-69 - Scanning failed with exit code 1. Stderr: I don't like the look of /home/dmr/workspace/matrix-content-scanner-python/perf/temp/matrix.org/MKYSaqghosWAaMkfOTGqAXWu
```